mono/mini/exceptions-amd64.c

```diff
@@ -71,6 +72,125 @@ static LONG CALLBACK seh_unhandled_exception_filter(EXCEPTION_POINTERS* ep)
 }
 
 /*
+* mono_win32_get_handle_stackoverflow (void):
+*
+* Returns a pointer to a method which restores the current context stack
+* and calls handle_exceptions, when done restores the original stack.
+*/
+static gpointer
+mono_win32_get_handle_stackoverflow (void)
+{
+	static guint8 *start = NULL;
```

Comment:
This function change a couple of nonvolatile registers that are not restored before exit. This can cause issues for callers depending on their usage and optimizations.